azure-pipelines.yml

```diff
@@ -1,342 +1,404 @@
-                  -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
-                  -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
-                  -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
+          - job: "DeployInfrastructuremis"
+            dependsOn: DeployInfrastructure
+            steps:
+              - template: pipeline-steps/deploy-service.yaml
+                parameters:
+                  environment: ${{ component.env }}
+                  location: ${{ parameters.location }}
+                  stack: "aks-mis"
+                  project: $(project)
+                  tfInitSub: ${{ variables.tfInitSub }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  action: ${{ parameters.action }}
+                  setClusterNumbersEnvVar: false
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_role_assignment.uami_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_read_rg["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
 
-  - stage: BootStrapClusters
-    displayName: 'BootStrap Clusters'
-    dependsOn: Aks
-    jobs:
-      - job: BootStrap
+      - stage: "BootStrapClusters_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: BootStrap Clusters"
+        dependsOn:
+          - "Aks_${{ component.env }}"
         condition: |
-          or(
-            and(succeeded(), eq('${{ parameters.action }}', 'apply')),
-            and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
-          )
+            or (
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                in('${{ parameters.cluster }}', '00', '01'),
+                eq(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], '${{ parameters.cluster }}')
+              ),
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                eq('${{ parameters.cluster }}', 'All')
+              ),
+              and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
+            )
```

Comment:
Do we want to change this? This is what's causing bootstrap to always run which we don't need, can you change it back to the original of :

```
              or(
                and(succeeded(), eq('${{ parameters.action }}', 'apply')),
                and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
              )
```

Or is there a reason it needs to have this similar aks logic?

Reply:
@NickAzureDevops

Reply:
Done. revert the condition to original main one